src/swupd/cmd/create-fullfiles/main.go

```diff
@@ -0,0 +1,89 @@
+package main
+
+// Creates the fullfiles for a specific version given a state
+// directory. Used to test the logic in the library functions.
+
+import (
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"path/filepath"
+	"runtime/pprof"
+
+	"swupd"
+)
+
+func usage() {
+	fmt.Fprintf(os.Stderr, `Usage: create-fullfiles [FLAGS] STATEDIR VERSION
+Flags:
+`)
+	flag.PrintDefaults()
+	os.Exit(2)
+}
+
+func main() {
+	log.SetFlags(0)
+
+	cpuProfile := flag.String("cpuprofile", "", "write CPU profile to a file")
+	outputDir := flag.String("o", "", "output directory, creates temporary if not specified")
+	flag.Usage = usage
+
+	if *cpuProfile != "" {
+		f, err := os.Create(*cpuProfile)
+		if err != nil {
+			log.Fatalf("couldn't create file for CPU profile: %s", err)
+		}
+		pprof.StartCPUProfile(f)
+		defer pprof.StopCPUProfile()
+	}
+
+	flag.Parse()
+
+	if flag.NArg() != 2 {
+		usage()
+	}
+
+	stateDir := flag.Arg(0)
+	version := flag.Arg(1)
+
+	chrootDir := filepath.Join(stateDir, "image", version, "full")
+	if _, err := os.Stat(chrootDir); err != nil {
+		log.Fatalf("couldn't access the full chroot: %s", err)
+	}
+
+	manifestFile := filepath.Join(stateDir, "www", version, "Manifest.full")
+	if _, err := os.Stat(manifestFile); err != nil {
+		log.Fatalf("couldn't access the full manifest: %s", err)
+	}
+
+	// TODO: Make ReadManifestFromFile return the Manifest.
+	m := &swupd.Manifest{}
+	err := m.ReadManifestFromFile(manifestFile)
+	if err != nil {
+		log.Fatalf("couldn't read full manifest %s: %s", manifestFile, err)
+	}
+
+	if *outputDir == "" {
+		tempDir, err := ioutil.TempDir("", "fullfiles-")
```

Comment:
Do we want the first param to be "" (which will make it use something like /tmp, or "." to use the current directory.

Reply:
In my testing, I wanted to avoid messing with the current directory, but don't have a strong opinion on the matter. I will change to use ".".